relayer/generic_relayer/pricing/src/app.ts

```diff
@@ -0,0 +1,97 @@
+import Koa from "koa";
+import Router from "koa-router";
+import {
+  Next,
+  RelayerApp,
+  StandardRelayerAppOpts,
+  StandardRelayerContext,
+  logging,
+  wallets,
+  missedVaas,
+  providers,
+  sourceTx,
+} from "relayer-engine";
+import { EVMChainId } from "@certusone/wormhole-sdk";
+import { processProviderPriceUpdate } from "./processor";
+import { Logger } from "winston";
+import deepCopy from "clone";
+import { loadAppConfig } from "@wormhole-foundation/offchain-generic-relayer/src/env";
+
+export type PricingContext = StandardRelayerContext & {
+  deliveryProviders: Record<EVMChainId, string>;
+  wormholeRelayers: Record<EVMChainId, string>;
+  opts: StandardRelayerAppOpts;
+};
+
+async function main() {
+  const { env, opts, deliveryProviders, wormholeRelayers } =
+    await loadAppConfig();
+  const logger = opts.logger!;
+  logger.debug("Redis config: ", opts.redis);
+
+  const app = new RelayerApp<PricingContext>(env, opts);
+  const { privateKeys, name, wormholeRpcs } = opts;
+
+  app.logger(logger);
+  app.use(logging(logger));
+  app.use(providers(opts.providers));
+  if (opts.privateKeys && Object.keys(opts.privateKeys).length) {
+    app.use(
+      wallets(env, {
+        logger,
+        namespace: name,
+        privateKeys: privateKeys!,
+        metrics: { registry: app.metricsRegistry },
+      })
+    );
+  }
+  if (opts.fetchSourceTxhash) {
+    app.use(sourceTx());
+  }
+
+  // Set up middleware
+  app.use(async (ctx: PricingContext, next: Next) => {
```

Comment:
How is this suppossed to be triggered? It appears to be listening for deliveryVaas and running the pricing based off that? What's the rationale here vs. a background job?

Reply:
It would be good in the future to trigger based off a cron timer or something similar, but currently the trigger isn't meaningful, I agree